### PR TITLE
channeldb: remove older migrations

### PR DESCRIPTION
**Pull request by joostjager**
To reduce maintenance on migration code, we move to only supporting upgrades from previous major releases.

### CODE REVIEW

File: channeldb/error.go [hunk 1]
```diff
@@ -14,6 +14,10 @@ var (
 	// prior database version.
 	ErrDBReversion = fmt.Errorf("channel db cannot revert to prior version")
 
+	// ErrDBVersionTooLow is returned when detecting an attempt to upgrade a
+	// version for which migration is no longer supported.
+	ErrDBVersionTooLow = fmt.Errorf("channel db version to old to upgrade")
```
Comment (halseth) [inline channeldb/error.go]: s/to/too

File: channeldb/db.go [hunk 2]
```diff
@@ -1125,6 +1127,13 @@ func (d *DB) syncVersions(versions []version) error {
 			latestVersion)
 		return ErrDBReversion
 
+	case meta.DbVersionNumber < minUpgradeVersion:
+		log.Errorf("Refusing to upgrade from db_version=%d to "+
+			"lower version=%d. Upgrade via intermediate major "+
```
Comment (halseth) [inline channeldb/db.go]: it is not really lower version though? Should be something like "Refusing to upgrade from too old db version=..."?

File: channeldb/db.go [hunk 3]
```diff
@@ -37,72 +37,6 @@ var (
 	// for retrieving all migration function that are need to apply to the
 	// current db.
 	dbVersions = []version{
-		{
```
Comment (halseth) [inline channeldb/db.go]: carnage! 😮

File: channeldb/db.go [hunk 4]
```diff
@@ -1168,6 +1176,21 @@ func getLatestDBVersion(versions []version) uint32 {
 	return versions[len(versions)-1].number
 }
 
+// getMinUpgradeVersion returns the minimum version required to upgrade the
+// database.
+func getMinUpgradeVersion(versions []version) uint32 {
```
Comment (cfromknecht) [inline channeldb/db.go]: not a blocker, but now that we're removing migrations, this logic should probably be able to handle the case where there are no migrations in a given release. would need a similar change to `getLatestDBVersion` as well

**Pull request merged**